Allow pagination with a primary key besides :id

trptcolin opened the pull request:
This one is more straightforward than the last one: Korma defaults to a primary key of `:id`, but it can be set to something else. 

There was one small edge case I handled: if there is _no_ primary key, I fall back to trying a column named `:id`, just as the existing behavior would've done. I can delete that fallback behavior if you'd prefer to fail fast if there's no primary key.

trptcolin committed:
Allow pagination with a primary key besides :id

kgann commented on src/cumin/pagination.clj:
```diff
@@ -22,13 +22,14 @@
-      (select)
-      (ffirst)
-      (last)))
+  (let [pk (get-in query [:ent :pk] :id)]
```
would `(aggregate (count :*) :count)` work just as well?

trptcolin replied:
Good question. I could've sworn I'd seen MySQL performing way slower w/ `count(*)` than w/ `count(id)` (years ago), but everything I'm seeing now says otherwise (and I don't use MySQL anymore anyway). Will update; that'll be more straightforward.

kgann replied:
I think using `:pk` is great but perhaps default to `:*` instead of assuming there is an `:id` key.

trptcolin replied:
Oops, I pushed too quick ;) Will update to use the `:pk` lookup and fall back to `:*`.

trptcolin committed:
Use COUNT(*) instead of COUNT(column)

These should be semantically identical (unless the column was nullable),
but the code is certainly clearer this way.

trptcolin committed:
Default to count by primary key, fall back to :*

kgann commented:
:+1: Thanks!
